src/amber/environment/settings.cr

```diff
@@ -10,6 +10,27 @@ module Amber::Environment
       skip: Array(String?),
       context: Array(String?))
 
+
+    struct SmtpSettings
+      property host = "127.0.0.1"
+      property port = 1025
+      property enabled = false
+      property username = ""
+      property password = ""
+      property tls = false
+
+      def self.from_hash(settings = {} of String => String | Int32 | Bool | Nil) : self
+        i = new
+        i.host     = settings["host"]?     ? settings["host"].as String     : i.host
+        i.port     = settings["port"]?     ? settings["port"].as Int32      : i.port
+        i.enabled  = settings["enabled"]?  ? settings["enabled"].as Bool    : i.enabled
+        i.username = settings["username"]? ? settings["username"].as String : i.username
+        i.password = settings["password"]? ? settings["password"].as String : i.password
+        i.tls      = settings["tls"]?      ? settings["tls"].as Bool        : i.tls
```

Comment:
@eliasjpr @elorest @drujensen if you have some ideas on how to make this cleaner, I'm open to suggestions.

Reply:
@robacarp Why not follow the same pattern as the LoggingType. Create an alias for a SmtpType with a `NamedTupple(host: String, port: Int32, enabled: Bool, username: String, Password: String, tls: Bool)`

```crystal
alias SmtpType = NamedTupple(
  host: String, 
  port: Int32, 
  enabled: Bool, 
  username: String, 
  password: String, 
  tls: Bool
)

property smtp : SmtpType? 

# Yaml mapping
smtp: { type:  SmtpType, getter: false, default: nil }
```

```yaml
...
smtp:
  host: "smtp.postmarkapp.com", 
  port: 25, 
  enabled: true, 
  username: "59df0bfb-7995-493c-bc6d-68sdf7g87f6d", 
  password: "59df0bfb-7995-493c-bc6d-68sdf7g87f6d", 
  tls: true
...
Or it can simply not be defined
```

This way you can avoid to have the smtp Struct

Reply:
@eliasjpr I initially tried to do that but that meant that I had to add _all_ the smtp fields or none. This way allows several different shapes of configurations:

Full: 

```yaml
smtp:
  host: "smtp.postmarkapp.com"
  port: 25
  enabled: true
  username: "59df0bfb-7995-493c-bc6d-68sdf7g87f6d"
  password: "59df0bfb-7995-493c-bc6d-68sdf7g87f6d"
  tls: true
```

Mailcatcher:

```yaml
smtp:
  host: "localhost"
  port: 1025
  enabled: true
```

Etc. Initially, I built an `SmtpType` following the pattern of Logger but it didn't allow optional sub-fields.

But it's also possible there was something else going on crashing the yaml builder.